test/unit/lib/challenge.test.ts

```diff
@@ -26,3 +27,130 @@ describe('Challenge Generation', (): void => {
     expect(Buffer.byteLength(challenge, 'base64')).toBe(8)
   })
 })
+
+describe('Signature Verification', (): void => {
+  // Repeat test with different keys
+  for (let index = 0; index < 4; index++) {
```

Comment:
I don't like wrapping `it` statements in loops like this. It makes it really hard to debug test failures. 

Perhaps internally you could generate a list of KeyPairs (`keyPairList`), and run a reducer function over these.

Reply:
beforeEach() might also be useful ?

Reply:
Will implement the change

Reply:
if you want to generate `tests` or `it` in a loop, you should also generate proper description, why using magical `4` instead of `7` ?